src/rust/src/padding.rs

```diff
@@ -110,6 +109,62 @@ impl PKCS7PaddingContext {
     }
 }
 
+#[pyo3::pyclass]
+pub(crate) struct ANSIX923PaddingContext {
+    block_size: usize,
+    buffer: Option<Vec<u8>>,
+}
+
+#[pyo3::pymethods]
+impl ANSIX923PaddingContext {
+    #[new]
+    pub(crate) fn new(block_size: usize) -> ANSIX923PaddingContext {
+        ANSIX923PaddingContext {
+            block_size: block_size / 8,
+            buffer: Some(Vec::new()),
+        }
+    }
+
+    pub(crate) fn update<'p>(
+        &mut self,
+        buf: CffiBuf<'p>,
+        py: pyo3::Python<'p>,
+    ) -> CryptographyResult<pyo3::Bound<'p, pyo3::types::PyBytes>> {
+        match self.buffer.as_mut() {
+            Some(v) => {
```

Comment:
I don't think any of this is necessary, if you look at pkcs7padder, we can jsut return the buffer and count the length when padding.